lib/closure_tree/hierarchy_maintenance.rb

```diff
@@ -35,10 +35,13 @@ def _ct_before_save
     end
 
     def _ct_after_save
-      if changes[_ct.parent_column_name] || @was_new_record
+      as_5_1 = ActiveSupport.version >= Gem::Version.new('5.1.0')
+      changes_method = as_5_1 ? :saved_changes : :changes
```

Comment:
Wow, that's pretty horrible that Rails is making us do this. Bummer.

I can't think of a more elegant way to do it though. Thanks for this PR.

Reply:
@mceachen I wonder, why are you flustrated? I take it as part of something large like Rails which need to evolve somehow forward. And this is something I missed in Rails and welcome it.